src/main/scala/zio/prelude/Equal.scala

```diff
@@ -300,6 +307,18 @@ object Equal extends Lawful[Equal] {
   implicit def EitherEqual[A: Equal, B: Equal]: Equal[Either[A, B]] =
     Equal[A] either Equal[B]
 
+  /**
+   * Derives an `Equal[Equal[A]]` given an `Enumerable[A]`.
+   */
+  implicit def EqualEqual[A: Enumerable]: Equal[Equal[A]] =
+    Equal.make { (l, r) =>
+      val pairs = for {
+        a1 <- Enumerable[A].enumerate
+        a2 <- Enumerable[A].enumerate
+      } yield l.equal(a1, a2) == r.equal(a1, a2)
```

Comment:
I think we want `a1.zip.a2(_ === _).forall(identity)`.

Reply:
Or maybe I misunderstand...

Reply:
I think ideally we want the Cartesian product here because we want to say that for all combinations of values the two equal instances give the same answer.